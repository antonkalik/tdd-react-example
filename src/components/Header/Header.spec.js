import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { Header } from "./index";
import { MemoryRouter } from "react-router-dom";

const header = (
  <MemoryRouter>
    <Header />
  </MemoryRouter>
);

describe("<Header />", () => {
  it("should render Header", () => {
    render(header);
    expect(screen.getByTestId("header")).toMatchSnapshot();
  });

  it("should have title", () => {
    const { container } = render(header);
    expect(container).toHaveTextContent("Logo Title");
  });

  it("should render controls in correct view", () => {
    render(
      <MemoryRouter initialEntries={["/cards"]}>
        <Header />
      </MemoryRouter>
    );
    const resetButton = screen.getByRole("button", { name: "Reset" });
    const addButton = screen.getByRole("button", { name: "Add" });
    expect(resetButton).toBeInTheDocument();
    expect(addButton).toBeInTheDocument();
  });

  it("should call onReset function", () => {
    const onReset = jest.fn();
    render(
      <MemoryRouter initialEntries={["/cards"]}>
        <Header onReset={onReset} />
      </MemoryRouter>
    );
    const element = screen.getByRole("button", { name: "Reset" });
    userEvent.click(element);
    expect(onReset).toHaveBeenCalled();
  });

  it("should call onAdd function", () => {
    const onAdd = jest.fn();
    render(
      <MemoryRouter initialEntries={["/cards"]}>
        <Header onAdd={onAdd} />
      </MemoryRouter>
    );
    const element = screen.getByRole("button", { name: "Add" });
    userEvent.click(element);
    expect(onAdd).toHaveBeenCalled();
  });
});
