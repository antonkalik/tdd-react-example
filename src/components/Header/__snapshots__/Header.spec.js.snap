// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 60px;
  background-color: #777;
}

.c0 .buttons {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .buttons button {
  margin-left: 10px;
  width: 5rem;
}

<header
  class="c0"
  data-testid="header"
>
  <a
    href="/"
  >
    <h1>
      Logo Title
    </h1>
  </a>
  <ul>
    <li>
      <a
        href="/cards"
      >
        Cards
      </a>
    </li>
    <li>
      <a
        href="/about"
      >
        About
      </a>
    </li>
  </ul>
</header>
`;
